components/playlist/browser/playlist_service.cc

```diff
@@ -672,6 +666,29 @@ void PlaylistService::DeleteAllPlaylistItems() {
   CleanUpOrphanedPlaylistItemDirs();
 }
 
+void PlaylistService::RemoveLocalDataForItem(
+    const mojom::PlaylistItemPtr& item) {
+  DCHECK(item);
+  if (!item->cached)
+    return;
+
+  item->cached = false;
+  DCHECK(item->media_source.is_valid()) << "media_source should be valid";
+  item->media_path = item->media_source;
```

Comment:
Curious why deleting `item->media_path` first and get `GetMediaPath()` later for deleting?
I think `item->media_path` could be used for deleting and we can set empty string to it?
and do we need to make `media_path` store `media_source` instead of empty?

Reply:
> Curious why deleting item->media_path first and get GetMediaPath() later for deleting?

Good call.

> do we need to make media_path store media_source instead of empty?

It was just to allow client to use only meida_path only when they want to load something. Otherwise, they're going to have to access goth media_path and media_source, when there's no cached media file.

Reply:
> Curious why deleting item->media_path first and get GetMediaPath() later for deleting?

@simonhong I tried this, but it seems this would be more complicated that the current implementation. We need to remove "file://" part and convert GURL to file path base on platform FILE_PATH_LITERAL. So I think it'd be better to leave as is. What do you think?

Reply:
Make sense to me. Let's use original one with above comment.

Reply:
Thanks! Reverted.